Terminal.Gui/Views/SpinnerView.cs

```diff
@@ -71,5 +298,1492 @@ protected override void Dispose (bool disposing)
 
 			base.Dispose (disposing);
 		}
+
+		/// <summary>
+		/// Gets/Sets the spinner view style based on the <see cref="Terminal.Gui.SpinnerStyle"/>
+		/// </summary>
+		public SpinnerStyle SpinnerStyle {
+			get => _style;
+			set {
+				_style = value;
+				_bounceReverse = false;
+				SpinDelayInMilliseconds = 80;
+				SpinBounce = false;
+				SpinReverse = false;
+				_hasSpecial = false;
+				switch (value) {
+				case SpinnerStyle.Custom: // Placeholder when user has specified delay and frames manually
```

Comment:
What would you say to using a simple class-heirarchy for all this instead of a enum + giant switch statement?

E.g.

```cs
public class SpinnerStyle {
   public int Delay { get ; set } = 80;
   public bool Bounce { get ; set } = false;
...
   public string [] Frames { get ; set; };

   /// location is screen relative like all primitive draw functions (see `Ruler`)
   public virtual void Draw (Point location) {
       // Code from `SpinnerView.Redraw()`
   }
}

public class Dots : SpinnerStyle {
   public Dots () {
        Frames = new string []
					{
						"⠋",
						"⠙",
						"⠹",
						"⠸",
						"⠼",
						"⠴",
						"⠦",
						"⠧",
						"⠇",
						"⠏",
					};
     }
}
...
```

Reply:
That would be a lot of classes in the root namespace.  

If going down this route I think they should either be nested classes or start with the same text e.g. `SpinnerStyleDots`, `SpinnerStyleBounce` etc.  I hate nested classes so probably the later?

Theres probably an alternative solution involving a dictionary, something like:

```csharp
this.dict = new Dictionary<SpinnerStyle,SpinnerDefinition> ()

this.Add(SpinnerStyle .Dots, new SpinnerDefinition(80,false,
"⠋","⠙","⠹","⠸","⠼","⠴","⠦","⠧","⠇","⠏",));
this.Add(SpinnerStyle .Star, new SpinnerDefinition(80,false,
/*etc*/));
this.Add(SpinnerStyle.Aesthetic, new SpinnerDefinition(80,false,
/*etc*/));
```

SpinnerDefinition would be something like 

```csharp
class SpinnerDefinition
{
	public int Delay { get; }
	public bool Bounce { get; }
	public string [] AnimationFrames { get; set; }
	public SpinnerDefinition (int delay, bool bounce, params string [] animationFrames)
	{
		Delay = delay;
		Bounce = bounce;
		AnimationFrames = animationFrames;
	}
}
```

Reply:
Re: "Frames," perhaps it could be called "Animation" or "Sequence" or even just "Spinner" instead?

Re: SpinnerWidth, I imagined it would be possible for the SpinnerView frame to be larger than the animation itself. But I suppose such a case could be accomplished with margins or padding.

Re: classes vs. enum/switch [vs. dictionary, etc.], as I mentioned in the comment to the PR, I was just trying to match the methodology I saw used for other styles. I feel like [nested] classes would be mo' better (especially here with so many possibilities), and I'm happy to change it [to whatever you guys decide. I might offer similar PRs for other styles, so it would be good to decide now. ;-) ]

Reply:
I think the current `enum` + `switch` is the most fragile, hard to test, and hard to extend of the choices.

I don't have @tznind aversion to nested classes (Thomas, would you mind explaining why you have said aversion?).

Reply:
Its a soft aversion 😉 and I have some in FileDialog so it's a bit hypocritical too.

Mostly its based on the reasons in the microsoft design guidlines:

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/nested-types

That said though, this is a good exception to the rule I think.  If I understand the pattern here we are talking about avoiding cluttering in the root namespace with 20+ new classes.  So thats worth it to me.

Lets go with the nested classes approach 🚀